### PR TITLE
Check max required open files for titan

### PR DESCRIPTION
**Pull request by sunxiaoguang**
Signed-off-by: Xiaoguang Sun <sunxiaoguang@zhihu.com>

## What have you changed? (mandatory)
Check max open files required by titan engine when it is enabled.

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
- Summarize your change (mandatory)
Add max open files required by titan engine when it is enabled  

## What are the type of the changes? (mandatory)
- Misc (other changes)

## How has this PR been tested? (mandatory)
Manually tested

## Does this PR affect documentation (docs) or release note? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

## Refer to a related PR or issue link (optional)
NA
## Benchmark result if necessary (optional)
NA
## Add a few positive/negative examples (optional)
NA

### CODE REVIEW

File: src/binutil/server.rs [hunk 1]
```diff
@@ -434,8 +434,12 @@ fn pre_start(cfg: &TiKvConfig) {
 }
 
 fn check_system_config(config: &TiKvConfig) {
+    let mut rocksdb_max_open_files = config.rocksdb.max_open_files;
+    if config.rocksdb.titan.enabled {
+        rocksdb_max_open_files *= 2;
```
Comment (siddontang) [inline src/binutil/server.rs]: please add a comment here

Comment (zhangjinpeng1987) [reply src/binutil/server.rs]: Is it possible that Titan has more files than RocksDB @Connor1996

Comment (sunxiaoguang) [reply src/binutil/server.rs]: Done. 

> please add a comment here

**Comment by zhangjinpeng1987** [comment]
Please sign-off

File: src/binutil/server.rs (hunk 1 above)
Comment (sunxiaoguang) [reply src/binutil/server.rs]: > Is it possible that Titan has more files than RocksDB @Connor1996

Looks like Titan uses the same max open files setup as rocksdb does. 

image

**Comment by sunxiaoguang** [comment]
> Please sign-off

Oops, It's there now.

**Pull request merged**